src/mixer/anderson_mixer.hpp

```diff
@@ -0,0 +1,180 @@
+// Copyright (c) 2013-2019 Simon Frasch, Anton Kozhevnikov, Thomas Schulthess
+// All rights reserved.
+//
+// Redistribution and use in source and binary forms, with or without modification, are permitted provided that
+// the following conditions are met:
+//
+// 1. Redistributions of source code must retain the above copyright notice, this list of conditions and the
+//    following disclaimer.
+// 2. Redistributions in binary form must reproduce the above copyright notice, this list of conditions
+//    and the following disclaimer in the documentation and/or other materials provided with the distribution.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY EXPRESS OR IMPLIED
+// WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A
+// PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR
+// ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+// PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+// CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR
+// OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+/** \file anderson_mixer.hpp
+ *
+ *   \brief Contains definition and implementation sirius::Anderson.
+ */
+
+#ifndef __Anderson_MIXER_HPP__
+#define __Anderson_MIXER_HPP__
+
+#include <tuple>
+#include <functional>
+#include <utility>
+#include <vector>
+#include <limits>
+#include <memory>
+#include <exception>
+#include <cmath>
+#include <numeric>
+
+#include "SDDK/memory.hpp"
+#include "mixer/mixer.hpp"
+#include "linalg/linalg.hpp"
+
+namespace sirius {
+namespace mixer {
+
+/// Anderson mixer.
+/** 
```

Comment:
This is a Doxygen comment, which allows for  LaTeX formulas. I would prefer to not have UTF characters here